mlflow/pyfunc/__init__.py

```diff
@@ -206,6 +206,100 @@
+or to override additional methods, you should use ``PythonModel``.
+
+Before looking at code examples, it's important to note that both methods are serialized via
+`cloudpickle <https://github.com/cloudpipe/cloudpickle>`_. cloudpickle can serialize Python
+functions, lambda functions, and locally defined classes and functions inside other functions. This
+makes cloudpickle especially useful for parallel and distributed computing where code objects need
+to be sent over network to execute on remote workers, which is a common deployment paradigm for
+MLflow.
+
+That said, cloudpickle has some limitations.
+
+- **Environment Dependency**: cloudpickle does not capture the full execution environment, so in
+    MLflow we must pass ``pip_requirements``, ``extra_pip_requirements``, or an ``input_example``,
+    the latter of which is used to infer environment dependencies.
+
+- **Object Support**: cloudpickle does not serialize objects outside of the Python data model.
+    Some relevant examples include raw files and database connections. If your program depends on
+    these, be sure to log ways to reference these objects along with your model.
+
+python_function
+###############
+If you're looking to serialize a simple python function without additional dependent methods, you
+can simply log a predict method via the keyword argument ``python_model``.
+
+.. code-block:: python
+    :caption: python_function Example
+
+    import mlflow
+    import pandas as pd
+
+    # Define a simple function to log
+    def predict(model_input):
+        return model_input.apply(lambda x: x * 2)
+
+    # Save the function as a model
+    with mlflow.start_run():
+        mlflow.pyfunc.log_model("model", python_model=predict, pip_requirements=["pandas"])
+        run_id = mlflow.active_run().info.run_id
+
+    # Load the model from the tracking server and perform inference
+    model = mlflow.pyfunc.load_model(f"runs:/{run_id}/model")
+    x_new = pd.Series([1,2,3])
+
+    print(f"Prediction:\n\t{model.predict(x_new)}")
+
+
+PythonModel
+###############
+If you're looking to serialize a more complex object, for instance a class that handles
+preprocessing, complex prediction logic, or custom serialization, you should use the
+``PythonModel`` class. MLflow has tutorials on building custom PyFunc models, as shown
+`here <https://mlflow.org/docs/latest/traditional-ml/creating-custom-pyfunc/index.html>`_,
+so instead of duplicating that information, in this example we'll recreate the above functionality
+to highlight the differences. Note that this PythonModel implementation is overly complex and
+should be a python_function instead.
+
+.. code-block:: python
+    :caption: PythonModel Example
+
+    import mlflow
+    import pandas as pd
+
+    class MyModel(mlflow.pyfunc.PythonModel):
+        def predict(self, context, model_input, params=None):
+            return model_input.apply(lambda x: x * 2)
```

Comment:
```suggestion
            return [x*2 for x in model_input]
```

super-nit: list comprehension over map (iirc our linter suggest similar to main code)